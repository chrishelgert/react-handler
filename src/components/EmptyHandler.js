import React, { PropTypes } from 'react';
import isEmpty from '../utils/isEmpty';

/**
 * This Mini-Handler handles only the empty-state
 * or display the given Component.
 */
const EmptyHandler = ({
  message,
  emptyPropertyKey,
  EmptyComponent,
  Component,
  componentProps = {},
}) => {
  if (isEmpty(emptyPropertyKey, componentProps)) {
    if (EmptyComponent) {
      return <EmptyComponent>{message}</EmptyComponent>;
    }

    return <div className="handler--empty">{message}</div>;
  }

  if (Component) {
    return <Component {...componentProps} />;
  }

  return null;
};

EmptyHandler.propTypes = {
  message: PropTypes.string.isRequired,
  emptyPropertyKey: PropTypes.string,
  EmptyComponent: PropTypes.func,
  Component: PropTypes.func,
  componentProps: PropTypes.object,
};

export default EmptyHandler;
