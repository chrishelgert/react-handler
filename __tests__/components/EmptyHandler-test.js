import React, { PropTypes } from 'react';
import renderer from 'react-test-renderer';
import EmptyHandler from '../../src/components/EmptyHandler';

const TestComponent = () => <h1>Test</h1>;
const EmptyComponent = ({ children }) => <h2 className="empty">{children}</h2>;
EmptyComponent.propTypes = { children: PropTypes.string };
const emptyPropertyKey = 'list';
const componentPropsFilled = { list: ['abc', 'def'] };
const componentPropsEmpty = { list: [] };
const emptyMsg = 'no results found';

describe('EmptyHandler', () => {
  it('should return the component when called without emptyPropertyKey', () => {
    const tree = renderer.create(
      <EmptyHandler
        Component={TestComponent}
        componentProps={componentPropsFilled}
        emptyMsg={emptyMsg}
      />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should return the component when the element in componentProps is not empty', () => {
    const tree = renderer.create(
      <EmptyHandler
        Component={TestComponent}
        componentProps={componentPropsFilled}
        emptyPropertyKey={emptyPropertyKey}
        emptyMsg={emptyMsg}
      />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should return the EmptyComponent, when the list is empty and called with EmptyComponent', () => {
    const tree = renderer.create(
      <EmptyHandler
        Component={TestComponent}
        EmptyComponent={EmptyComponent}
        componentProps={componentPropsEmpty}
        emptyPropertyKey={emptyPropertyKey}
        emptyMsg={emptyMsg}
      />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should return the EmptyMessage, when the list is empty', () => {
    const tree = renderer.create(
      <EmptyHandler
        Component={TestComponent}
        componentProps={componentPropsEmpty}
        emptyPropertyKey={emptyPropertyKey}
        emptyMsg={emptyMsg}
      />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
