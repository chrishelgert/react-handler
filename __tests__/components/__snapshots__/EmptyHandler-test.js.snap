// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyHandler returns null when called without Component and is not empty 1`] = `null`;

exports[`EmptyHandler returns the EmptyComponent, when the list is empty and called with EmptyComponent 1`] = `
<h2
  className="empty"
>
  no results found
</h2>
`;

exports[`EmptyHandler returns the EmptyMessage, when called without checkedProperty 1`] = `
<div
  className="handler--empty"
>
  no results found
</div>
`;

exports[`EmptyHandler returns the EmptyMessage, when the list is empty 1`] = `
<div
  className="handler--empty"
>
  no results found
</div>
`;

exports[`EmptyHandler returns the component when the element in componentProps is not empty 1`] = `
<h1>
  Test
</h1>
`;
